File: fxgl/src/main/java/com/almasb/fxgl/gameplay/notification/NotificationService.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.gameplay.notification;

import com.almasb.fxgl.gameplay.AchievementEvent;
import com.almasb.fxgl.gameplay.AchievementListener;
import com.almasb.fxgl.gameplay.AchievementProgressEvent;
import com.almasb.fxgl.ui.Position;
import javafx.scene.paint.Color;

/**
 * Notification service allows to push notifications.
 * This is a globally available service with globally visible notifications.
 *
 * @author Almas Baimagambetov (AlmasB) (almaslvl@gmail.com)
 */
public interface NotificationService extends AchievementListener {

    /**
     * Push a notification with given message.
     *
     * @param message the message
     */
    void pushNotification(String message);

    /**
     *
     * @return notification position
     */
    Position getPosition();

    /**
     * Set position of future notifications.
     *
     * @param position where to show notification
     */
    void setPosition(Position position);

    /**
     *
     * @return current background color for notifications
     */
    Color getBackgroundColor();

    /**
     * Set background color of notifications.
     *
     * @param backgroundColor the color
     */
    void setBackgroundColor(Color backgroundColor);

    /**
     * @return current text color for notifications
     */
    Color getTextColor();

    /**
     * Set text color of notifications.
     *
     * @param textColor the color
     */
    void setTextColor(Color textColor);

    void onAchievementEvent(AchievementEvent event);
}

File: fxgl-base/src/main/kotlin/com/almasb/fxgl/gameplay/notification/NotificationService.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.gameplay.notification

import com.almasb.fxgl.gameplay.achievement.AchievementListener
import com.almasb.fxgl.ui.Position
import javafx.scene.paint.Color

/**
 * Notification service allows to push notifications.
 * This is a globally available service with globally visible notifications.
 *
 * @author Almas Baimagambetov (AlmasB) (almaslvl@gmail.com)
 */
interface NotificationService : AchievementListener {

    /**
     *
     * @return notification position
     */
    /**
     * Set position of future notifications.
     *
     * @param position where to show notification
     */
    var position: Position

    /**
     *
     * @return current background color for notifications
     */
    /**
     * Set background color of notifications.
     *
     * @param backgroundColor the color
     */
    var backgroundColor: Color

    /**
     * @return current text color for notifications
     */
    /**
     * Set text color of notifications.
     *
     * @param textColor the color
     */
    var textColor: Color

    /**
     * Push a notification with given message.
     *
     * @param message the message
     */
    fun pushNotification(message: String)
}